Integrations/DigitalDefense/DigitalDefense.py

```diff
@@ -0,0 +1,596 @@
+            continue
+        vuln_event = {}
+        vuln_event['vuln-id'] = vuln.get('id')
+        vuln_event['hostname'] = vuln.get('hostname')
+        vuln_event['ip-address'] = vuln.get('ip_address')
+        vuln_event['port'] = vuln.get("port")
+        vuln_event['scan-id'] = vuln.get('scan_id')
+        vuln_event['vuln-title'] = vuln.get('title')
+        vuln_event['date-created'] = vuln.get('active_view_date_created')
+        vuln_event['ddi-severity'] = vuln['severities']['ddi']
+        vuln_event['vuln-info'] = vuln.get('data')
+
+        incident = {
+            'name': vuln.get('title'),
+            'occurred': vuln.get('active_view_date_created'),
+            'details': vuln.get('data'),
+            'rawJSON': json.dumps(vuln_event)
+        }
+        incidents.append(incident)
+    return incidents
+
+
+def fetch_incidents():
+    try:
+        last_run = demisto.getLastRun()
+        # Check if last_run exists and has a start_time, else create new start_time timestamp:
+        if last_run and last_run.get('start_time', False):
+            start_time = last_run.get('start_time')
+            # Check if user selected fetch frequency is within last_run[start_time]:
+            fetch_frequency_td = get_fetch_frequency_td()
+            if ((datetime.utcnow() - datetime.utcfromtimestamp(int(start_time))) < fetch_frequency_td):
+                demisto.debug("FrontlineVM fetch_incident -- frequency (" + str(fetch_frequency_td) +") within last start_time, sending empty incidents.")
+                demisto.setLastRun({'start_time': last_run.get('start_time')})
+                demisto.incidents([])
+                return
+            else:
+                demisto.debug("FrontlineVM fetch_incident -- frequency (" + str(fetch_frequency_td) +") exceeds last start_time, continuing to fetch incident events.")
+        else:
+            now = datetime.utcnow()
+            start_time = now.strftime("%s")
+        start_time_dt = datetime.utcfromtimestamp(int(start_time))
+        start_time_str = start_time_dt.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        demisto.debug("FrontlineVM fetch_incident -- continuing with start_time of " + str(start_time_str))
+
+        # Fetch vulnerabilities:
+        req_params = {}
+        req_params['lte_vuln_severity_ddi'] = str(INCIDENT_VULN_SEVERITY)
+        req_params['gte_vuln_date_created'] = start_time_str
+        req_params = parse_params(req_params)
+        vulns = get_fvm_data(VULN_ENDPOINT, API_TOKEN, params=req_params)
+
+        # Create incidents
+        if vulns:
+            demisto.debug("FrontlineVM fetch_incidents -- getting all incidents.")
+            incidents = get_vuln_incidents(vulns, start_time_dt)
+        else:
+            demisto.debug("FrontlineVM fetch_incidents -- no new vulnerabilities found, no incidents created.")
+            incidents = []
+
+        now = datetime.utcnow()
+        start_time = now.strftime("%s")
```

Comment:
Seconds will not matter when fetching vulnerabilities with our API. The API will pull all vulnerabilities for that day given the date, regardless of seconds. This is why there is a condition to check every vulnerability pulled for that start_time:
`if (av_date_created_dt < start_time_dt): continue`
else:
if the vulnerability date created timestamp is greater than start_time, then it will be converted into an fetch incident event.